File: src/main/scala/net/machinemuse/powersuits/item/ItemPowerArmorBoots.scala
package net.machinemuse.powersuits.item

import ic2.api.item.IMetalArmor
import net.machinemuse.utils.render.MuseRenderer
import net.minecraft.entity.EntityLivingBase
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.inventory.EntityEquipmentSlot
import net.minecraft.item.ItemStack
import net.minecraftforge.fml.common.Optional

@Optional.InterfaceList (Array(
  new Optional.Interface (iface = "ic2.api.item.IMetalArmor", modid = "IC2", striprefs = true)
) )
class ItemPowerArmorBoots extends ItemPowerArmor(0, EntityEquipmentSlot.FEET) with IMetalArmor{
  val iconpath = MuseRenderer.ICON_PREFIX + "armorfeet"

  setUnlocalizedName("powerArmorBoots")

  override def isMetalArmor(itemStack: ItemStack, entityPlayer: EntityPlayer): Boolean = true

  override def protectEntity(entityLivingBase: EntityLivingBase, itemStack: ItemStack, s: String, b: Boolean): Boolean = ???
}

File: src/main/java/net/machinemuse/powersuits/item/ItemPowerArmorBoots.java
package net.machinemuse.powersuits.item;

import cpw.mods.fml.common.Optional;
import cpw.mods.fml.relauncher.Side;
import cpw.mods.fml.relauncher.SideOnly;
import ic2.api.item.IMetalArmor;
import net.machinemuse.utils.render.MuseRenderer;
import net.minecraft.client.renderer.texture.IIconRegister;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.item.ItemStack;

/**
 * Ported to Java by lehjr on 10/26/16.
 */
@Optional.InterfaceList ({
        @Optional.Interface (iface = "ic2.api.item.IMetalArmor", modid = "IC2", striprefs = true)
})
public class ItemPowerArmorBoots extends ItemPowerArmor implements IMetalArmor{
    public final String iconpath = MuseRenderer.ICON_PREFIX + "armorfeet";

    public ItemPowerArmorBoots() {
        super(0, 3);
        setUnlocalizedName("powerArmorBoots");
    }

    @SideOnly(Side.CLIENT)
    @Override
    public void registerIcons(IIconRegister iconRegister) {
        itemIcon = iconRegister.registerIcon(iconpath);
    }

    @Override
    public boolean isMetalArmor(ItemStack itemStack, EntityPlayer entityPlayer) {
        return true;
    }
}
